docs/GUIDE_FOR_KOGITO_DEVS.md

```diff
@@ -0,0 +1,352 @@
+
+The YAML to deploy the application varies depending on the specific 
+requirements of it. In all cases, you deploy the YAML using 
+the command `oc apply -f [YAML file name]`.
+
+#### Simple
+In the simplest case, no `KogitoInfra` or `KogitoSupportingService` 
+are required. For example, [process-quarkus-example](https://github.com/kiegroup/kogito-examples/tree/stable/process-quarkus-example) 
+without persistence/events enabled. The YAML for this case 
+can be found in 
+[`examples/process-quarkus-example-runtime.yaml`](https://github.com/kiegroup/kogito-operator/blob/master/examples/process-quarkus-example-runtime.yaml) 
+in the operator:
+```yaml
+apiVersion: app.kiegroup.org/v1beta1
+kind: KogitoRuntime
+metadata:
+  name: process-quarkus-example
+spec:
+  replicas: 1
+  image: quay.io/[Quay username]/process-quarkus-example
+```
+
+#### `KogitoInfra`
+An example of deploying with a `KogitoInfra` would deploying 
+the [process-quarkus-example](https://github.com/kiegroup/kogito-examples/tree/stable/process-quarkus-example) 
+with persistence enabled. The YAML for this case 
+can be found in 
+[`examples/process-quarkus-example-persistence-nobuild.yaml`](https://github.com/kiegroup/kogito-operator/blob/master/examples/process-quarkus-example-persistence-nobuild.yaml) 
+in the operator:
+```yaml
+# Infinispan operator should be pre-installed in namespace
+# And have installed an Infinispan server named "kogito-infinispan" in the same namespace of the Kogito resources
+# Follow these instructions to setup the Infinispan server:
+# https://infinispan.org/infinispan-operator/master/operator.html#creating_minimal_clusters-start
+apiVersion: app.kiegroup.org/v1beta1
+kind: KogitoInfra
+metadata:
+  name: kogito-infinispan-infra
+spec:
+  resource:
+    apiVersion: infinispan.org/v1
+    kind: Infinispan
+    name: kogito-infinispan
+---
+apiVersion: app.kiegroup.org/v1beta1
+kind: KogitoRuntime
+metadata:
+  name: example-quarkus
+spec:
+  # see the quarkus-jvm.Dockerfile in this directory to know how to build your image locally
+  image: quay.io/<namespace>/process-quarkus-example:latest
+  infra:
+    - kogito-infinispan-infra
+```
+
+Ensure that the Infinispan operator is installed in the namespace as 
+the comments in the YAML mention, and the image is built 
+with the `-Ppersistence` flag.
+
+# Operator Folder Structure
+The operator repository is composed of many folders. This section will cover the 
+general purpose of each:
+
+<!-- TODO: ask about unknown folders' purposes -->
+- `api/` - Definition for custom Kogito resources/types.
+- `bundle/` - ?
+- `cmd/` - Code for Kogito CLI.
+- `config/` - ?
+- `controllers/` - Code for operator's controllers (a controller watches a specific resource type).
+- `core/` - Bulk of operator's code.
+- `docs/` - Docs for Kogito developers.
+- `examples/` - Example YAML files for deployment onto Kubernetes.
+- `hack/` - Various shell scripts for `make`.
+- `internal/` - ?
+- `meta/` - ?
```

Comment:
good question, this seems to be where you configure resources that will be watched by operator.
@vaibhavjainwiz  ideas on what would be a good description?

Reply:
Maybe something like `Contains Scheme and type registrations used by Kubernetes client`?